### ISSUE TITLE
As a user, I can manage identities in the new Playground UI

### ISSUE THREAD

**Issue by EdMoffatt** [hyperledger/composer#533]
Functionality to issue Identities and add them to wallet needs to be implemented in the new UI.  Need to support a user switching between multiple Identities, using them to invite other participants in, and revoking Identities.

---

Although designs below include additional capabilities, the plan is to initially provide functional equivalence with the old UI with follow on work to expand that functionality. See also #649 for some additional Angular components to support current designs.

Tests required (latest code is available on [533-identity-page](https://github.com/jthub/hyperledger-composer/tree/533-identity-page) branch):
- [x] Add Identity Component - **James**
- [x] Identity Component - **Caroline**
- [x] Identity Issued Component - **Caroline**
- [x] Issue Identity Component - **Nick**
- [x] Identity Service - **Jake**
- [x] Wallet Service - **James**

**Comment by EdMoffatt** [comment]
Designs now added to Box here: https://ibm.box.com/s/4hplks51zz14o5rfze6hd5hyiwis9xct

Assigning to @caroline-church to work out who should take this one on :)

**Comment by EdMoffatt** [comment]
Haha, got a bit ahead of myself there - forgot what I'd attached was just low-fis XD

Assigning @winslet instead, as we really need the hi-fis first ;)

**Comment by EdMoffatt** [comment]
Updated low-fis based on recent discussions with @sstone1 @winslet and @Tobias-Hunter 

Online: https://xd.adobe.com/view/f478e722-20a6-485c-9d23-57ecfa507f61/

XD file on Box: https://ibm.box.com/s/yys8hbdlksj35vuy0tzsuj0an7wgx0pi

**Comment by winslet** [comment]
Hifi's incoming....

I would suggest clicking through Ed's mockup at the same time to give these screens context & a story. 

### Navbar dropdown showing Wallet ID's and link to Identity Registry

![fabric composer playground define 2 copy 112](https://cloud.githubusercontent.com/assets/13980028/24305084/a9387d8a-10b3-11e7-8c52-afd068b71cc8.png)

![fabric composer playground define 2 copy 110](https://cloud.githubusercontent.com/assets/13980028/24305074/a0994bc8-10b3-11e7-8f09-749496291e3c.png)

### Identity Registry page

![fabric composer playground define 2 copy 103](https://cloud.githubusercontent.com/assets/13980028/24305112/bb26125a-10b3-11e7-98ae-124624fd20d4.png)

#### Registry page with hover options shown

![fabric composer playground define 2 copy 101](https://cloud.githubusercontent.com/assets/13980028/24305123/ca4659f2-10b3-11e7-96bc-e6df5335c5c0.png)

### Issue new ID flow

_User selects "issue new id" from top right and modal appears_
![fabric composer playground define 2 copy 102](https://cloud.githubusercontent.com/assets/13980028/24305138/d9ef193e-10b3-11e7-96f4-81d8f0cc2586.png)

_They can enter an ID name and then a participant from the dropdown_
![fabric composer playground define 2 copy 104](https://cloud.githubusercontent.com/assets/13980028/24305206/130f166a-10b4-11e7-8109-2ff7e51f2003.png)

_Dropdown is also an autocomplete object similar to that used in Watson Conversation, this screen shows loading of participants on typing_
![fabric composer playground define 2 copy 104](https://cloud.githubusercontent.com/assets/13980028/24305233/27427e24-10b4-11e7-9532-136ba4bddb14.png)

eg
![image](https://cloud.githubusercontent.com/assets/13980028/24305247/33d3a8ca-10b4-11e7-9841-20ca328533a0.png)
![image](https://cloud.githubusercontent.com/assets/13980028/24305251/37438a34-10b4-11e7-90f0-a80f7e7c006a.png)

_User selects a participant and continues_
![fabric composer playground define 2 copy 109](https://cloud.githubusercontent.com/assets/13980028/24305306/60eadc5c-10b4-11e7-8ac2-ba13ced4c3cb.png)

_User selects "allow this id to issue new ID's" then clicks next_
![fabric composer playground define 2 copy 105](https://cloud.githubusercontent.com/assets/13980028/24305324/74cc618c-10b4-11e7-99f1-0ea5fd60b9a2.png)

_On the next screen they are presented with the secret to be shown once and a number of options. At this stage if they select "Done" the close icon top right or "Add to my wallet" the modal will close and the ID will appear in the table. The copy icons copy the relevant text to the users clipboard_
![fabric composer playground define 2 copy 106](https://cloud.githubusercontent.com/assets/13980028/24305393/b24e3f26-10b4-11e7-9843-1d29d687b9e6.png)

_User clicks Done and see's the ID appear in the registry_
![fabric composer playground define 2 copy 107](https://cloud.githubusercontent.com/assets/13980028/24305409/c045962e-10b4-11e7-8f02-27a34cf43555.png)

**Marketch**
[Identity_Management_FC_SW_1.zip](https://github.com/fabric-composer/fabric-composer/files/868653/Identity_Management_FC_SW_1.zip)

Any questions please let me know, of if I need to explain anything. Thanks!!

**Comment by EdMoffatt** [comment]
Worth a quuuuuuick note to say: whilst Sam's HiFis always show "Admin" next to the dropdown arrow (i.e. as the thing you click to bring up the menu), this will actually display the ID that the user is currently using.
For example, in the second screenshot shown, "Mr Auctioneer 1979" would actually be what's shown there, not "Admin".

(For those super interested in the details, the reason it's admin in the mocks is that we switched it and forgot to switch it back!  Because we were most worried that people might not understand it was a sort of "user profile dropdown thingy" if it said "Admin" on it, so we wanted to show a few people the mockups with "Admin" in and ask them what they thought that bit of the UI was and what they'd see if they clicked it - results were everyone getting it, so we stuck with it!)

**Comment by EdMoffatt** [comment]
As part of a GitHub "lets get super organised" drive...

- Converting this to an Epic, as there are a few child Issues being worked as part of it
- Changing to the Playback 4 milestone, so that we can track the current playback tasks in ZenHub (ZenHub doesn't play so well with Projects!)

I'm also going to raise a new issue to reflect the bit of splintered-off work that's currently blocking progress on this one.  Will link to that here in a second...!

**Comment by EdMoffatt** [comment]
Feel this issue should explicitly state...

We're not going to have the Identity Registry yet, and therefore some modifications to the designed behaviour are required.

@jt-nti - would it be possible to attach some screengrabs of the proposed implementation we're landing on here, so that they're available for discussion from the community?  Anyone feeding back on the latest attached designs will be feeding back on some more distant-future work than they might think at the moment ;)

**Comment by jt-nti** [comment]
Current designs (as demoed on [last community call]9https://github.com/hyperledger/composer/wiki/Meeting-27th-April-20170)):

![image](https://cloud.githubusercontent.com/assets/3535067/25626774/7b584710-2f58-11e7-839f-df97cc3a0724.png)

![image](https://cloud.githubusercontent.com/assets/3535067/25627108/7307df34-2f59-11e7-9e48-427f942d3f99.png)

![image](https://cloud.githubusercontent.com/assets/3535067/25627135/8a64ac02-2f59-11e7-9412-ee854b3c5fa0.png)

**Comment by EdMoffatt** [comment]
This is looking sweet, @jt-nti :)  There are a couple of things I don't see, which I don't think are blocked by the Identity Registry not being available yet...

- [ ] Display name of current connection profile next to the globe icon
- [ ] Instead of the person icon, display the name of the current ID and a little "dropdown carat" (a filled triangle pointing down)
- [ ] Have the dropdown that opens on clicking the ID name, displaying IDs available in this wallet, which can be picked from, or then the option to "Manage Identities" at the bottom which accesses the screens shown in your first screenshot

There are also some visual design points that we should clear up before we close out this one:

- [ ] The "Identity Issued" screen is looking kinda "no heirarchy" at the moment.  I know we don't yet have the ability to generate the shortlink and email (do we have an issue to find an alternative way of doing this and add it back in?) so we can't exactly create @winslet 's design yet, but it feels like it'd be worth Sam and James sitting together and tweaking this one.  What I mean by no heirarchy is that there's really important info for the user to take from that page, but nothing's standing out - I think it's easy to skim past at the moment, and in Sam's hi-fi I didn't feel that!
- [ ] The dropdowns look to be working great!  But the presentation of participants is not so nice.  We should pretty that up to match the designs, so that just the participant name is displayed first, and then the participant type next to that in differently styled text, as per the mockups.  Again, it's not just pretty, it's making the most important information for making the decision of which one to pick clear to the user when they're faced with the decision :)

**Comment by jt-nti** [comment]
There is definitely some follow on work once there's something working. I think the first three items should be covered by number 4 on #649.

The participant is just a typeahead at the moment. As well as adding the planned dropdown behaviour (number 2 on #649) just displaying the name and unqualified type could be ambiguous. Perhaps that's an edge case but I do think being able to paste in the fully qualified participant ID as text is useful.

The identity issued screen could definitely do with a bit of tweaking (just noticed it's missing the warning icon as well). It would be good to get that done this week if Sam has some time, unfortunately this issue is blocked by a build problem at the moment :(

**Comment by jt-nti** [comment]
@markcastle3 I had a chat with @EdMoffatt and I think we can close this issue off after I've added the current id to the top nav bar
